modules/dns/dns_api.rb

```diff
@@ -15,11 +16,22 @@ class Api < ::Sinatra::Base
       type = params[:type].upcase unless params[:type].nil?
 
       log_halt(400, "'create' requires fqdn, value, and type parameters") if fqdn.nil? || value.nil? || type.nil?
-      log_halt(400, "unrecognized 'type' parameter: #{type}") unless type == 'A' || type == 'PTR'
+      validate_dns_name(fqdn)
 
       begin
-        server.create_a_record(fqdn, value) if type == 'A'
-        server.create_ptr_record(fqdn, value) if type == 'PTR'
+        case type
+        when 'A'
+          ip = IPAddr.new(value, Socket::AF_INET).to_s
```

Comment:
I'm ok with changing the internal API. I'm not aware of any non-core dns providers other than power-dns.